tensorflow/compiler/mlir/tools/kernel_gen/transforms/tf_to_jit_invocations.cc

```diff
@@ -211,51 +225,143 @@ struct PackJITCompileOpPattern
 struct TFToJITInvocationPass
     : public TFToJITInvocationPassBase<TFToJITInvocationPass> {
   void getDependentDialects(DialectRegistry &registry) const override {
-    registry.insert<mlir::kernel_gen::tf_framework::TFFrameworkDialect>();
+    registry.insert<mlir::kernel_gen::tf_framework::TFFrameworkDialect, 
+    scf::SCFDialect, shape::ShapeDialect>();
   }
   explicit TFToJITInvocationPass(llvm::ArrayRef<int64_t> tile_sizes,
                                  llvm::ArrayRef<int64_t> unroll_factors,
                                  int64_t max_supported_rank, bool enable_ftz,
-                                 bool cpu_codegen) {
+                                 bool index_64bit, bool cpu_codegen,
+                                 bool jit_i64_indexed_for_large_tensors) {
     tile_sizes_ = tile_sizes;
     unroll_factors_ = unroll_factors;
     max_supported_rank_ = max_supported_rank;
     enable_ftz_ = enable_ftz;
+    index_64bit_ = index_64bit;
     cpu_codegen_ = cpu_codegen;
+    jit_i64_indexed_for_large_tensors_ = jit_i64_indexed_for_large_tensors;
   }
 
   void runOnFunction() override {
     MLIRContext *ctx = &getContext();
     RewritePatternSet patterns(ctx);
-    PopulateTFToJITInvocationPatterns(ctx, &patterns, tile_sizes_,
-                                      unroll_factors_, max_supported_rank_,
-                                      enable_ftz_, cpu_codegen_);
+    if(jit_i64_indexed_for_large_tensors_)
+    {
+      Populate64BitIndexerPatterns(
```

Comment:
As you're passing all the parameters to the populate functions anyways, we could also make them one and merge `Populate64BitIndexerPatterns` into `PopulateTFToJITInvocationPatterns`. This would nicely align with the pass then.

Reply:
Merged as requested.